docs/core/config/mailbox/mail_location.md

```diff
@@ -27,30 +27,24 @@ See [[variable]] for a full list, but the most commonly used ones are:
 
 | Variable | Description |
 | -------- | ----------- |
-| `%u` | Full username. |
-| `%n` | User part in `user@domain`; same as `%u` if there's no domain. |
-| `%d` | Domain part in `user@domain`; empty if there's no domain. |
+| `%{user}` | Full username. |
+| `%{user | username}` | User part in `user@domain`; same as `%{user}` if there's no domain. |
+| `%{user | domain}` | Domain part in `user@domain`; empty if there's no domain. |
 
 ### Directory Hashing
 
-You can use three different kinds of hashes in [[variable]].
+Examples on how to do it:
 
-* `%N` is MD5-based "new hash" which works similarly to `%H` except it
-  gives more uniform results.
+* How to replace '%N' in new format:
 
-  * Example: `%2.256N` would return maximum 256 different hashes in range
+  * %2.256N becomes `%{ user | md5 | fold(256) | hex(2)}` would return maximum 256 different hashes in range
     `00..ff`.
```

Comment:
Have you tested that this results in exactly the same output? If it doesn't, we break existing mail location paths. Also the old code allowed eg. `%30Nu` to get values 0..29. People might actually be using these. Would it still work with new code?

Also previously `%256Nu` gave values like 0, 1, 2, 3, .. values while `%2.256Nu` gave 00, 01, 02, 03, ... values. Can this work?

The fold() seems to be pretty much designed to replace this %N modifier. Maybe it could be changed to support all these directly without the need for |hex?